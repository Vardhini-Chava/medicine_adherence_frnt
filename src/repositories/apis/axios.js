import defaultAxios from 'axios';
import { API_URL } from '../var';

const axios = defaultAxios.create({
  baseURL: `${API_URL}`,
  headers: {'Content-Type': 'application/json'},
});

defaultAxios.interceptors.request.use(
  (request) => {
    return request;
  },
  (error) => {
    return Promise.reject(error);
  }
);

export default axios;
