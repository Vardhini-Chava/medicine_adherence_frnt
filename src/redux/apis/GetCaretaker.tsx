import axios from './Axiosdeclaration';

async function fetchcaretaker(payload) {
  console.log('payload', payload);
  const response = await axios.get(`/api/v1/caretakers?patientId=${payload}`);
  console.log(response + ' Saga ');
  return response.data;
}

export default fetchcaretaker;
