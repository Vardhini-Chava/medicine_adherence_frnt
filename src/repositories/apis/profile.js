import axios from './axios';
import { API_URLS } from '../apiUrl';
export const profile = {
    saveProfile: async function saveprofile(payload) {
        const response = await axios.put(`${API_URLS.SAVE_PROFILE}?userId=${payload}`);
        return response.data;
      }
}

axios.interceptors.request.use(
  (request) => {
    return request;
  },
  (error) => {
    return Promise.reject(error);
  }
);