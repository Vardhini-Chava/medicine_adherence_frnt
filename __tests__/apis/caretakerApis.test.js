import { careTaker } from "../../src/repositories/apis/careTaker";

describe("test caretaker apis", () => {
  it("test caretaker", () => {
    const payload = "payload";
    expect(careTaker.caretaker(payload)).toEqual({
      _U: 0,
      _V: 0,
      _W: null,
      _X: null,
    });
  });

  it("test emailcaretaker", () => {
    const payload = "payload";
    expect(careTaker.emailcaretaker(payload)).toEqual({
      _U: 0,
      _V: 0,
      _W: null,
      _X: null,
    });
  });

  it("test reqcaretaker", () => {
    const payload = "payload";
    expect(careTaker.reqCaretaker(payload)).toEqual({
      _U: 0,
      _V: 0,
      _W: null,
      _X: null,
    });
  });

  it("test sendimage", async () => {
    const payload = "payload";
    const response = { data: {} };
    const result = await careTaker.sendImage(payload);
    expect(result).toEqual(response);
  });
});
