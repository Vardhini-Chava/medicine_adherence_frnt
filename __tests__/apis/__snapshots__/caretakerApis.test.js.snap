// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test caretaker apis test sendimage 1`] = `
Promise {
  "_U": 0,
  "_V": 0,
  "_W": null,
  "_X": null,
}
`;
